### ISSUE TITLE
make test error with 0.5.0 on php 7.4.4

### ISSUE THREAD

**Issue by kpcyrd** [jvoisin/snuffleupagus#322]
Running make test with the latest release tarball (0.5.0) on arch linux with php 7.4.4 gives me the following test failures:
```
=====================================================================
TIME END 2020-03-26 14:54:03

=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :    1
---------------------------------------------------------------------

Number of tests :  297               260
Tests skipped   :   37 ( 12.5%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    6 (  2.0%) (  2.3%)
Expected fail   :    2 (  0.7%) (  0.8%)
Tests passed    :  252 ( 84.8%) ( 96.9%)
---------------------------------------------------------------------
Time taken      :   15 seconds
=====================================================================

=====================================================================
EXPECTED FAILED TEST SUMMARY
---------------------------------------------------------------------
Disable functions - die [tests/disable_function/disabled_functions_die.phpt]  XFAIL REASON: 
Disable functions - exit [tests/disable_function/disabled_functions_exit.phpt]  XFAIL REASON: 
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Disable functions - Ensure that client certificates validation can't be disabled [tests/disable_function/disabled_function_ensure_client_valid_certs.phpt]
Disable functions - Ensure that client certificates validation can't be disabled via `curl_multi_setopt` [tests/disable_function/disabled_function_ensure_client_valid_certs_curl_multi_setopt.phpt]
Disable functions - Ensure that client certificates validation can't be disabled via `curl_setopt_array` [tests/disable_function/disabled_function_ensure_client_valid_certs_curl_setopt_array.phpt]
Disable functions - Ensure that server certificates validation can't be disabled [tests/disable_function/disabled_function_ensure_server_valid_certs.phpt]
Disable functions - Ensure that server certificates validation can't be disabled via `curl_multi_setopt` [tests/disable_function/disabled_function_ensure_server_valid_certs_curl_multi_setopt.phpt]
Disable functions - Ensure that server certificates validation can't be disabled via `curl_setopt_array` [tests/disable_function/disabled_function_ensure_server_valid_certs_curl_setopt_array.phpt]
=====================================================================
make: *** [Makefile:131: test] Error 1
```

**Comment by jvoisin** [comment]
Can you try to install `php-curl` to see if it fixes your problem?

**Comment by kpcyrd** [comment]
Hm, the `php` package in arch linux seems to contain the curl.so module:
```
% pacman -Ql php | grep curl
php /usr/lib/php/modules/curl.so
```
You can reproduce this with the following PKGBUILD and running `extra-x86_64-build`:
```
# Maintainer: kpcyrd <kpcyrd[at]archlinux[dot]org>
# Contributor: Julien (jvoisin) Voisin <julien.voisin+snuffleupagus@dustri.org>

pkgname=php-snuffleupagus
_pkgname=snuffleupagus
pkgver=0.5.0
pkgrel=1
pkgdesc="Security module for PHP"
url='https://snuffleupagus.readthedocs.io/'
arch=('x86_64')
license=('LGPL3')
depends=('php' 'php-fpm')
source=(https://github.com/jvoisin/${_pkgname}/archive/v${pkgver}/${_pkgname}-${pkgver}.tar.gz)
b2sums=('a74b307e1050b48896f661d31d24deb29856e81c73dcac7670e4abbd5cd139188acee47a9a372c797fc32eff5b593df12d34d1f801c8e87ff54cc089f5171fb6')

build() {
  cd ${_pkgname}-${pkgver}/src
  phpize
  ./configure --prefix=/usr --enable-"${_pkgname}"
  make
}

check() {
  cd ${_pkgname}-${pkgver}/src
  NO_INTERACTION=1 REPORT_EXIT_STATUS=1 make test
}

package() {
  cd ${_pkgname}-${pkgver}/src
  make INSTALL_ROOT="${pkgdir}" install
  install -D -m644 "config/default.rules" "${pkgdir}/etc/php/conf.d/${_pkgname}.rules"
  install -D -m644 "config/${_pkgname}.ini" "${pkgdir}/etc/php/conf.d/${_pkgname}.ini"
}

# vim: ts=2 sw=2 et:
```

**Comment by jvoisin** [comment]
I don't have an archlinux box handy to test this unfortunately, but I'll try to allocate some time this week-end.

In the meantime, can you post the content of the `.out` and `.diff` files of the failed tests please?

**Comment by jvoisin** [comment]
I can reproduce this locally: it seems that php 7.4 doesn't load the `curl` extension correctly, and explicitly specifying it in the phpt file results in a command line looking like this:

```
/usr/bin/php7.4  -n -c '/home/jvoisin/dev/snuffleupagus/src/tmp-php.ini'  -d "extension=Could not startup.
/bz2.so" -d "extension=snuffleupagus.so" -d "output_handler=" -d "open_basedir=" -d "disable_functions=" -d "output_buffering=Off" -d "error_reporting=32767" -d "display_errors=1" -d "display_startup_errors=1" -d "log_errors=0" -d "html_errors=0" -d "track_errors=0" -d "report_memleaks=1" -d "report_zend_debug=0" -d "docref_root=" -d "docref_ext=.html" -d "error_prepend_string=" -d "error_append_string=" -d "auto_prepend_file=" -d "auto_append_file=" -d "ignore_repeated_errors=0" -d "precision=14" -d "memory_limit=128M" -d "log_errors_max_len=0" -d "opcache.fast_shutdown=0" -d "opcache.file_update_protection=0" -d "opcache.revalidate_freq=0" -d "zend.assertions=1" -d "zend.exception_ignore_args=0" -d "extension_dir=/home/jvoisin/dev/snuffleupagus/src/modules/" -d "sp.configuration_file=/home/jvoisin/dev/snuffleupagus/src/tests/disable_function/config/disabled_function_curl_verify_certs.ini" -f "/home/jvoisin/dev/snuffleupagus/src/tests/disable_function/disabled_function_ensure_client_valid_certs.php"  2>&1
```

I spent some time trying to fix it, but didn't even managed to understand why phpt's preprocessor couldn't start.

**Comment by jvoisin** [comment]
This should be fixed in 3e2b6545b623f0ba1b016c03b43d999cfcbc8d2f
